source/renderer/app/features/menu-updater/makeReactionCallback.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import { matchPath } from 'react-router-dom';
+import { WalletSettingsStateEnum } from '../../../../common/ipc/api';
+import { ROUTES } from '../../routes-config';
+import type { MakeReactionCallbackArgs } from './types';
+
+const walletRoutesExceptSettings = Object.values(ROUTES.WALLETS).filter(
+  (route) => route !== ROUTES.WALLETS.SETTINGS
+);
+
+const makeReactionCallback = ({
+  stores: { profile, router, uiDialogs },
+  rebuildApplicationMenu,
+}: MakeReactionCallbackArgs) => () => {
+  let walletSettingsState = WalletSettingsStateEnum.hidden;
+
+  const itIsTheWalletSettingsPage = matchPath(router.location?.pathname, {
+    path: ROUTES.WALLETS.SETTINGS,
+  })?.isExact;
+  const walletSettingsOptionVisible = itIsTheWalletSettingsPage
+    ? false
+    : walletRoutesExceptSettings.some(
+        (path) => matchPath(router.location?.pathname, { path })?.isExact
+      );
+
+  if (walletSettingsOptionVisible) {
+    walletSettingsState = uiDialogs.activeDialog
+      ? WalletSettingsStateEnum.disabled
+      : WalletSettingsStateEnum.enabled;
+  }
+
+  if (profile.currentLocale) {
+    // Just mentioning profile.currentLocale to make the Reaction watch its changes
+  }
```

Comment:
That is not an elegant solution at all, however I had no other idea probably because I am new in the mobx world.

Reply:
Maybe you could decorate [MenuUpdater](https://github.com/input-output-hk/daedalus/pull/2838/files#diff-cf32b35793444c25a614edfc40ddbf30f62d3a2181b84bc0f4b9bf954b1962d4R9)  with `observer`

Reply:
The problem is that I don't need the value of the `profile.currentLocale` (the main app obtains that value by itself), but I need to react on its changes. Event if this logic would be placed in the react component decorated with `observer` as I understand I also would need to do the same in order to tell the mobx that I want to react on changes of that property. But I could be wrong since I am new to mobx. I will play around :+1: